packages/backend-function/src/factory.ts

```diff
@@ -234,3 +293,31 @@ const nodeVersionMap: Record<NodeVersion, Runtime> = {
   18: Runtime.NODEJS_18_X,
   20: Runtime.NODEJS_20_X,
 };
+
+const secretPathSuffix = '_PATH';
+const secretPlaceholderText = '<value will be resolved during runtime>';
+const secretPaths: string[] = [];
+
+const translateEnvironmentProp = (
+  functionEnvironmentProp: HydratedFunctionProps['environment'],
+  backendSecretResolver: BackendSecretResolver
+): Record<string, string> => {
+  const result: Record<string, string> = {};
+  const secretPathEnvVars = [];
+
+  for (const [key, value] of Object.entries(functionEnvironmentProp)) {
+    if (typeof value !== 'string') {
+      const secretPath = backendSecretResolver.resolveToPath(value);
+      result[key + secretPathSuffix] = secretPath;
```

Comment:
instead of putting each path in a separate env var, we could just store a single env var that contains a JSON stringified record of `env var: path`. That would also eliminate the need for the `SECRET_PATH_ENV_VARS` below (or rather it would replace it).